packages/wallet-manager/src/wallets/Evm/Evm.ts

```diff
@@ -0,0 +1,164 @@
+import { ethers } from 'ethers';
+import events from 'events';
+import type {
+  Web3Provider,
+  ExternalProvider,
+  Provider
+} from '@ethersproject/providers';
+import { customEVMEvents, IEvmWallet } from '../../interfaces';
+import { AddChain } from '../../types';
+import { checkWindow } from '../../utils';
+
+class EvmWallet extends events.EventEmitter implements IEvmWallet {
+  address?: string;
+  signer?: ethers.Signer;
+  web3Provider!: Web3Provider;
+
+  constructor(provider?: Web3Provider) {
+    super();
+
+    if (!window.ethereum) {
+      throw new Error('window.ethereum is not defined.');
+    }
+    if (provider) {
+      this.web3Provider = provider;
+    } else {
+      this.web3Provider = new ethers.providers.Web3Provider(
+        window.ethereum as ExternalProvider
+      );
+    }
+    this.appendProviderEvents();
+  }
+
+  /**
+   * @name calculateAccountData
+   * @param accounts
+   * @returns {Promise<void>}
+   */
+  private async resetAccounts(accounts?: string[]): Promise<void> {
```

Comment:
do you really need these JSDoc with ts?

also, name is not right 😆

Reply:
I have removed the js docs for almost all the code except the context one that has an example on usage.

Any suggestion for the name of the method?

Reply:
nono i meant `@name` isn't the same as the name of the method

Reply:
ok, JsDocs are removed so I will resolve this convo